Documentation/building/linux-instructions.md

```diff
@@ -99,36 +102,57 @@ D:\git\coreclr> build.cmd linuxmscorlib
 
-Build the Framework Native Components
-======================================
+Build the Framework
+===================
 
 ```
-ellismg@linux:~/git/corefx$ src/Native/build.sh
-ellismg@linux:~/git/corefx$ cp bin/Linux.x64.Debug/Native/*.so ~/coreclr-demo/runtime
+ellismg@linux:~/git/corefx$ ./build.sh
 ```
 
-Build the Framework Managed Components
-======================================
+For the purposes of Hello World, you need to copy a few required files to the demo folder.
 
-We don't _yet_ have support for building managed code on Linux, so you'll need a Windows machine with clones of CoreFX project.
+```
+ellismg@linux:~/git/corefx$ cp bin/Linux.x64.Debug/Native/*.so ~/coreclr-demo/runtime
+ellismg@linux:~/git/corefx$ cp bin/Linux.AnyCPU.Debug/System.Console/System.Console.dll ~/coreclr-demo/runtime
+ellismg@linux:~/git/corefx$ cp bin/Linux.AnyCPU.Debug/System.Diagnostics.Debug/System.Diagnostics.Debug.dll ~/coreclr-demo/runtime
+```
 
-You will build the rest of the framework that out of the corefx repository.  
-You need to pass some special parameters to build.cmd when building out of the CoreFX repository.
+The runtime directory should now look like the following:
 
 ```
-D:\git\corefx> build.cmd /p:OSGroup=Linux /p:SkipTests=true
+matell@linux:~$ ls ~/coreclr-demo/runtime/
+corerun             System.Diagnostics.Debug.dll
+libcoreclr.so       System.Native.so
+mscorlib.dll        System.Net.Http.Native.so
+System.Console.dll  System.Security.Cryptography.Native.so
 ```
 
-It's also possible to add `/t:rebuild` to the build.cmd to force it to delete the previously built assemblies.
+(Alternative) Build the Framework Managed Components on Windows
+===============================================================
 
-For the purposes of Hello World, you need to copy over both `bin\Linux.AnyCPU.Debug\System.Console\System.Console.dll` and `bin\Linux.AnyCPU.Debug\System.Diagnostics.Debug\System.Diagnostics.Debug.dll`  into the runtime folder on Linux. (e.g `~/coreclr-demo/runtime`).
+If the build fails (for example due to Mono issues), alternatively you can build it on Windows.
+You'll need a Windows machine with a clone of the CoreFX repository.
 
-After you've done these steps, the runtime directory on Linux should look like this:
+Build the Native Components on Linux
+------------------------------------
 
 ```
-matell@linux:~$ ls ~/coreclr-demo/runtime/
-corerun  libcoreclr.so  mscorlib.dll  System.Console.dll  System.Diagnostics.Debug.dll System.Native.so
+ellismg@linux:~/git/corefx$ src/Native/build.sh
+ellismg@linux:~/git/corefx$ cp bin/Linux.x64.Debug/Native/*.so ~/coreclr-demo/runtime
+```
+
+Build the Managed Components on Windows
+---------------------------------------
+
+You need to pass some special parameters to build.cmd when building out of the CoreFx repository for Linux.
+
 ```
+D:\git\corefx> build.cmd /p:OSGroup=Linux /p:SkipTests=true
+```
```

Comment:
I think cross-platform building choice is better, but this is a very rare use-case where user would prefer building on Windows rather than the target OS itself. Most people are interested in doing everything on the target OS without needing a Windows machine at all. Especially now, when users have another choice to download the latest pre-build artifacts from the CI: http://dotnet-ci.cloudapp.net/job/dotnet_corefx_linux_debug/lastSuccessfulBuild/artifact/. 

I have collected my notes on FreeBSD in the form of gist guides:

[Building CoreCLR on FreeBSD VM](https://gist.github.com/jasonwilliams200OK/06db121ec534821fe541)
[Building CoreFX on FreeBSD VM](https://gist.github.com/jasonwilliams200OK/6efa7907e66275df2d24)

(we can probably put this kind of guides/walkthroughs under something like `Documentation/building/community-experiences` which may grow in number in future, when CoreCLR will start building on Solaris, Gentoo, DragonFly etc.)